crates/aptos-node-identity/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{format_err, Result};
+use aptos_types::{chain_id::ChainId, PeerId};
+use once_cell::sync::OnceCell;
+use std::sync::Arc;
+
+/// The global [AptosNodeIdentity]
+static APTOS_NODE_IDENTITY: OnceCell<Arc<AptosNodeIdentity>> = OnceCell::new();
+
+/// Structure that holds information related to a node's identity
+pub struct AptosNodeIdentity {
+    pub chain_id: OnceCell<ChainId>,
+    pub peer_id: Option<PeerId>,
+    pub peer_id_str: Option<String>,
+    pub hostname: Option<String>,
+}
+
+/// Initializes the [AptosNodeIdentity] using the provided [NodeConfig] and
+/// sets it globally exactly once.
+pub fn init(peer_id: Option<PeerId>) -> Result<()> {
+    let identity = AptosNodeIdentity {
+        chain_id: OnceCell::new(),
+        peer_id,
+        peer_id_str: peer_id.map(|id| id.to_string()),
+        hostname: hostname::get()
+            .ok()
+            .and_then(|name| name.into_string().ok()),
+    };
+
+    APTOS_NODE_IDENTITY
+        .set(Arc::new(identity))
+        .map_err(|_| format_err!("APTOS_NODE_IDENTITY was already set"))
+}
+
+pub fn set_chain_id(chain_id: ChainId) -> Result<()> {
```

Comment:
nit: add a comment to this function? Everyone other function has a comment (it feels lonely) 😸